CMakeLists.txt

```diff
@@ -1,177 +1,225 @@
+  COMMAND ${CMAKE_COMMAND} -E make_directory ${PROJECT_SOURCE_DIR}/assets/updater/
+  COMMAND ${CMAKE_COMMAND} -E copy ${PROJECT_BINARY_DIR}/${${SHORT_NAME}_UPDATER_SFO_NAME} ${PROJECT_SOURCE_DIR}/assets/updater/param.sfo
+  COMMENT "Copying built updater sfo to assets"
+  DEPENDS ${${SHORT_NAME}_UPDATER_SFO_NAME}
+  DEPENDS ${PROJECT_BINARY_DIR}/${${SHORT_NAME}_UPDATER_SFO_NAME}
+)
+
+option(DEBUGNET "Enable debugnet for logging" ON)
+
+# Default build type
+if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+  set(CMAKE_BUILD_TYPE Debug CACHE STRING "Default build" FORCE)
+  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release") # For CMake GUI
+endif()
+
+
+find_package(Git)
+if(NOT Git_FOUND)
+  message(WARNING "Git not found, using unknown as tag...")
+  add_definitions(-DGIT_COMMIT="unknown")
+else()
+  execute_process(
+          COMMAND ${GIT_EXECUTABLE} rev-parse HEAD
+          WORKING_DIRECTORY ${PROJECT_SOURCE_DIR}
+          OUTPUT_VARIABLE DGIT_COMMIT
+          OUTPUT_STRIP_TRAILING_WHITESPACE
+          ERROR_QUIET
+  )
+
+  add_definitions(-DGIT_COMMIT="${DGIT_COMMIT}")
+endif()
+
+
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wno-sign-compare -Wno-unused-parameter -std=c++11")
+
+#set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -g -Og")
+#set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -g -Og")
+
+#set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -Os")
+#set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -Os")
+
+find_package(CURL REQUIRED)
+find_package(OpenSSL REQUIRED)
+pkg_check_modules(CURLPP REQUIRED curlpp)
+
+include_directories(
+  src/
+)
+
+FUNCTION(ADD_RESOURCES out_var)
+  SET(result)
+  FOREACH(in_f ${ARGN})
+    SET(out_f "${CMAKE_CURRENT_BINARY_DIR}/${in_f}.o")
+    GET_FILENAME_COMPONENT(out_dir ${out_f} DIRECTORY)
+    ADD_CUSTOM_COMMAND(OUTPUT ${out_f}
+      COMMAND ${CMAKE_COMMAND} -E make_directory ${out_dir}
+      COMMAND ${CMAKE_LINKER} -r -b binary -o ${out_f} ${in_f}
+      DEPENDS ${in_f}
+      WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+      COMMENT "Building resource ${out_f}"
+      VERBATIM
+      )
+    LIST(APPEND result ${out_f})
+  ENDFOREACH()
+  SET(${out_var} "${result}" PARENT_SCOPE)
+ENDFUNCTION()
+
+file(GLOB_RECURSE res_files RELATIVE ${CMAKE_SOURCE_DIR}
+  assets/head.bin
+  assets/*.png
+  assets/*.jpeg
+  assets/*.yml
+  assets/*.ttf
+  assets/*.wav
+  assets/*.ogg)
+set(res_files ${res_files} assets/updater/eboot.bin assets/updater/param.sfo)
+
+add_resources(PROJECT_RESOURCES ${res_files})
+
+file(GLOB_RECURSE PROJECT_SOURCE_FILES "src/*.h" "src/*.hpp" "src/*.cpp" "src/*.c")
+
+set(VITA_ELF_NAME ${SHORT_NAME}.elf)
+add_executable(${VITA_ELF_NAME}
+  ${PROJECT_RESOURCES}
+  ${PROJECT_SOURCE_FILES}
+)
+
+target_compile_definitions(${VITA_ELF_NAME}
+  PRIVATE
+  VITA_VERSION="${VITA_VERSION}"
+  VHBB_SHORT_NAME="${SHORT_NAME}"
+  PACKAGE_TEMP_FOLDER=std::string\(\"${PACKAGE_TEMP_FOLDER}\"\)

```

Comment:
Why not just `"${PACKAGE_TEMP_FOLDER}"`?
I'd prefer we stay consistent.

Reply:
It's too late for that (staying consistent). I refactored it out from some `.cpp` or `.h` file (where the macro was a `std::string`. Generally I don't like the fact that some macros are `std::string`s. But it was mixed all over the code base (even before this PR). I didn't want to break compatibility. This is why I made it a `std::string` in the `CMakeLists` as well. I could go over the code base and replace all `std::string`s in macros with ordinary `"…"`s, if you are ok with that.

Reply:
True, the `char*` / `std::string` mix isn't great. It feels weird to me to see a class instantiation in a CMake file. Feel free to change just this occurrence, or all the occurrences if you prefer.